### PR TITLE
Increase button hit area

### PR DESCRIPTION
**Pull request by danielVebman**
# Overview

Increased the touch area for navigation bar buttons. 

I'll wait to merge until after Reade does. **Edit:** never mind.

# Changes Made

- Increased touch area for navigation bar buttons
   - Added custom navigation bar class
   - Overrode `hitTest`
- Edited permission alert text so it lays out better visually
- Renamed extension files to be e.g. `UIColor+Extension`

<sub>*Note: When I merged I got all those old commits. No clue how to get rid of them. Getting kind of annoying...*</sub>

# Test Coverage

Tested on device.

# Related Issues

#26

### CODE REVIEW

File: CourseGrab/ViewControllers/Login/LoginViewController.swift [hunk 1]
```diff
@@ -53,6 +54,10 @@ class LoginViewController: UIViewController {
             make.top.equalTo(logoView.snp.bottom)
         }
     }
+
+    override func viewWillAppear(_ animated: Bool) {
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/Login/LoginViewController.swift]: call `super.viewWillAppear(animated)` here as well

File: CourseGrab/ViewControllers/MainNavigationController.swift [hunk 2]
```diff
@@ -30,10 +30,51 @@ class InteractivePopRecognizer: NSObject, UIGestureRecognizerDelegate {
...
+
+    weak var navigationController: UINavigationController?
+
+    override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
+        if bounds.insetBy(dx: 0, dy: -tapOffset).contains(point), let item = navigationController?.topViewController?.navigationItem {
+            var candidateViews = [
+                item.backBarButtonItem?.customView,
+                item.leftBarButtonItem?.customView,
+                item.rightBarButtonItem?.customView
+                ].compactMap { $0 }
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/MainNavigationController.swift]: indentation seems off here

File: CourseGrab/ViewControllers/MainNavigationController.swift [hunk 3]
```diff
@@ -30,10 +30,51 @@ class InteractivePopRecognizer: NSObject, UIGestureRecognizerDelegate {
...
+            item.leftBarButtonItems?.compactMap { $0.customView }.forEach { candidateViews.append($0) }
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/MainNavigationController.swift]: personal preference but i think this is a bit more readable:
```
item.leftBarButtonItems?
  .compactMap { $0.customView }
  .forEach { candidateViews.append($0) }
```

File: CourseGrab/ViewControllers/MainNavigationController.swift [hunk 4]
```diff
@@ -30,10 +30,51 @@ class InteractivePopRecognizer: NSObject, UIGestureRecognizerDelegate {
...
+            item.rightBarButtonItems?.compactMap { $0.customView }.forEach { candidateViews.append($0) }
+
+            for view in candidateViews {
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/MainNavigationController.swift]: can we use `forEach` here? just to be consistent w/ above

**Comment by kevinchan159** [comment]
also did you make sure to branch off of your `master` branch? the commits that appear seem really old

File: CourseGrab/ViewControllers/MainNavigationController.swift (hunk 4 above)
Comment (danielVebman) [reply CourseGrab/ViewControllers/MainNavigationController.swift]: That does not work because you cannot return a non-void value in a forEach closure.

Comment (kevinchan159) [reply CourseGrab/ViewControllers/MainNavigationController.swift]: oh mb i didn't see that you were returning a value

**Pull request merged**